### PR TITLE
lakeFSFS: IAM Role Support Login call

### LINKED ISSUE #7658: lakeFSFS: IAM Role Support Login call

### PR DESCRIPTION
**Pull request by Isan-Rivkin**
Closes #7658
Docs in a separate PR #7656 (WIP)

### CODE REVIEW

**Comment by Isan-Rivkin** [comment]
Hey @arielshaqed , docs in a separate PR (WIP #7656)

File: clients/hadoopfs/src/main/java/io/lakefs/auth/AWSLakeFSTokenProvider.java [hunk 1]
```diff
@@ -94,53 +106,48 @@ private boolean needsNewToken() {
...
-         2. this.lakeFSAuthToken = call api.ExternalPrincipalLogin(identityToken, <lakeFS Token optional TTL>)
-        */
-        // dummy initiation
-        this.lakeFSAuthToken = new AuthenticationToken();
-        this.lakeFSAuthToken.setTokenExpiration(System.currentTimeMillis() + 60);
+
+        // build lakeFS login request
+        ExternalLoginInformation req = new ExternalLoginInformation();
+
+        if (this.lakeFSTokenTTLSeconds.isPresent()) {
```
Comment (arielshaqed) [inline clients/hadoopfs/src/main/java/io/lakefs/auth/AWSLakeFSTokenProvider.java:140]: Could use [this](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#flatMap-java.util.function.Function-) to apply an operator directly if present.

**Review by arielshaqed — APPROVED** [review]
Thanks!
How are we going to test this on a real lakeFS server? We need to integrate it properly.

**Comment by Isan-Rivkin** [comment]
@arielshaqed regarding testing, as of right now Iv'e been doing a lot of manuall testing with lakeFS / pyspark / fluffy. 
The challenge with proper integration test is that we need fluffy here, so I'll defer this into another issue on enterprise side.

**Pull request merged**